cmake/UseCcache.cmake

```diff
@@ -0,0 +1,59 @@
+# Compiling ccache via ccache is fine, because this uses a stable version which
+# is installed on the system.
+
+# Modified version of Craig Scott's "Professional CMake: A Practical Guide", 7th Edition
+function(use_ccache)
+  if(NOT CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+    message(STATUS " use_ccache() disabled, as it is not called from the project top level")
+    return()
+  endif()
+
+  find_program(CCACHE_PROGRAM ccache)
+  if(NOT CCACHE_PROGRAM)
+    message(STATUS " use_ccache() disabled, as ccache program not found")
+    return()
+  endif()
+
+  message(STATUS " use_ccache() enabled")
+
+  # This will override any environment settings, but not user settings in config file
```

Comment:
Isn't this incorrect? Environment variables have higher priority than config file settings.